### PR TITLE
Add mappings and readme

### PR DESCRIPTION
**Pull request by pbishwakarma**
This PR adds:

Mappings between the following objects and the dandi metadata schema:
- Specimen inventory
- Project inventory
- File manifest

### CODE REVIEW

File: design/mappings/file_manifest_mappings.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+fileManifest:
+  Filename: asset.path
+  Filetype: [
+    asset.path, #Split and get file extension
+    asset.encodingFormat, #Follows application/<filetype> format but type is just string
+    dandiset.metadata.assetsSummary.dataStandard # has RRID + standard name but is at data collection level
```
Comment (satra) [inline design/mappings/file_manifest_mappings.yml:6]: a data standard is not necessarily applicable at the file level. for example, with BIDS at the file level the data standards are NIFTI/JSON/etc.,., will BIDS is a collection level standard.

however, for NWB, indeed it's both an encoding format and a standard. we are trying to figure out more complex MIME types, such `application/vnd+hdf5+nwb` but formal use of such things is still under discussion.

File: design/mappings/file_manifest_mappings.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  ]
+  Sample ID: asset.path #Split to remove id from filename
+  Project: [
+    asset.wasGeneratedBy, #schemaKey=Project Not as reliable
+    dandiset.metadata.wasGeneratedBy, #schemaKey=Project More reliable
```
Comment (satra) [inline design/mappings/file_manifest_mappings.yml:11]: this is correct. an asset can belong to multiple collections. so deriving project from collection is appropriate.

File: design/mappings/file_manifest_mappings.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  ]
+  Data Collection: dandiset.metadata.name
```
Comment (satra) [inline design/mappings/file_manifest_mappings.yml:13]:
```suggestion
  Data Collection: dandiset.name
```

**Comment by satra** [comment]
thanks for doing all the mapping. looks very good overall. left some comments.

File: design/mappings/project_inventory_mappings.yml [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      # awardNumber linked to data collection and multiple are listed
+      # roleName=dcite:Funder used to distinguish funding organization vs contributing person
+    has use of results limited by: 
+      license: dandiset.metadata.license
+    has methodology specification: 
+      protocol: dandiset.metadata.protocol #protocol.io links
+    has output:
+      collection: ""
+    has contact person: 
+      person: dandiset.metadata.contributor #Required
```
[the remark this answers is not in the document]
Comment (pbishwakarma) [reply design/mappings/project_inventory_mappings.yml]: Thanks - updated

File: design/mappings/file_manifest_mappings.yml (hunk 1 above)
Comment (pbishwakarma) [reply design/mappings/file_manifest_mappings.yml]: That makes sense - removed dataStandard from this mapping. Mime type seems like the sensible field for this.

File: design/mappings/file_manifest_mappings.yml (hunk 3 above)
Comment (pbishwakarma) [reply design/mappings/file_manifest_mappings.yml]: Makes sense, updated the mappings to remove the metadata prefix

File: design/mappings/project_inventory_mappings.yml [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    is part of:
+      subprogram: 
+    has information resource:
+      Web resource:
+    is funded by: 
+      grant: dandiset.metadata.contributor.awardNumber
+      # awardNumber linked to data collection and multiple are listed
+      # roleName=dcite:Funder used to distinguish funding organization vs contributing person
+    has use of results limited by: 
+      license: dandiset.metadata.license
```
[the remark this answers is not in the document]
Comment (pbishwakarma) [reply design/mappings/project_inventory_mappings.yml]: This is based on the relations in the current schema - there are some redundant relations like this and the protocol links to project. Will review these soon when we work on schema definitions

**Review by carolth — APPROVED** [review]

**Pull request merged**